assets/js/components/wp-dashboard/WPDashboardPopularPagesGA4.stories.js

```diff
@@ -0,0 +1,157 @@
+	</WithRegistrySetup>
+);
+
+function provideGatheringData( registry, value ) {
+	registry.dispatch( MODULES_ANALYTICS_4 ).receiveIsGatheringData( value );
+}
+
+export const Ready = Template.bind( {} );
+Ready.storyName = 'Ready';
+Ready.args = {
+	setupRegistry: ( registry ) => {
+		const options = WPDashboardPopularPagesGA4.selectReportArgs(
+			registry.select
+		);
+
+		provideGatheringData( registry, false );
+		provideAnalytics4MockReport( registry, options );
+	},
+};
+
+export const GatheringData = Template.bind( {} );
+GatheringData.storyName = 'Gathering Data';
+GatheringData.args = {
+	setupRegistry: ( registry ) => {
+		provideGatheringData( registry, true );
+	},
+};
+
+export const ZeroData = Template.bind( {} );
+ZeroData.storyName = 'Zero Data';
+ZeroData.args = {
+	setupRegistry: ( registry ) => {
+		const options = WPDashboardPopularPagesGA4.selectReportArgs(
+			registry.select
+		);
+		const report = getAnalytics4MockResponse( options );
+		const zeroReport =
+			replaceValuesInAnalytics4ReportWithZeroData( report );
+		registry.dispatch( MODULES_ANALYTICS_4 ).receiveGetReport( zeroReport, {
+			options,
+		} );
+		provideGatheringData( registry, false );
+	},
+};
+
+export const Loading = Template.bind( {} );
+Loading.storyName = 'Loading';
+Loading.args = {
+	setupRegistry: ( registry ) => {
+		const options = WPDashboardPopularPagesGA4.selectReportArgs(
+			registry.select
+		);
+		provideAnalytics4MockReport( registry, options );
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.startResolution( 'getReport', [ options ] );
+	},
+};
+
+export const Error = Template.bind( {} );
+Error.storyName = 'Error';
+Error.args = {
+	setupRegistry: ( registry ) => {
+		provideModules( registry ); // "Data error in {module name}"
+		const options = WPDashboardPopularPagesGA4.selectReportArgs(
+			registry.select
+		);
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.receiveIsGatheringData( false );
+		const error = {
+			code: 'missing_required_param',
+			message: 'Request parameter is empty: metrics.',
+			data: {},
+		};
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.receiveError( error, 'getReport', [ options ] );
+		registry
+			.dispatch( MODULES_ANALYTICS_4 )
+			.finishResolution( 'getReport', [ options ] );
+	},
+};
+
+export default {
+	title: 'Views/WPDashboardApp/WPDashboardPopularPagesGA4',
+	decorators: [
+		( Story ) => (
+			<WithRegistrySetup
+				func={ ( registry ) =>
+					registry
+						.dispatch( CORE_USER )
+						.setReferenceDate( '2021-01-28' )
+				}
+			>
+				<Story />
+			</WithRegistrySetup>
+		),
+		( Story ) => (
+			<div id="dashboard-widgets">
+				<div
+					id="google_dashboard_widget"
+					className="postbox"
+					style={ { maxWidth: '600px' } }
+				>
+					<div className="inside">
+						<div className="googlesitekit-plugin">
+							<div className="googlesitekit-wp-dashboard">
+								<Story />
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		),
+	],
```

Comment:
This is also won't be needed because we can use `widgetDecorators` from the `common.stories.js`:
```suggestion
	decorators: widgetDecorators,
```